web.go

```diff
@@ -10,6 +10,10 @@ import (
 	"net/http"
 )
 
+var (
+	webConfiguration *Configuration
```

Comment:
Yeah, I'm not using the API so I didn't test it, sorry for that.

The reason I split the API (webConfiguration) out  was because other providers, like marathon, will overwrite any changes to the marathon config on next update.If the API has changed any settings in marathon managed config, they will be overwritten.

One option would be to split the API in two segments, one for informational purposes that would represent the whole config, and one segment that would allow updates to a subset of the config (API managed) .  I'm not sufficiently familiar with how traefik is being used today by others so I can't tell if that would be a major change or not.

Another option would be to change the API to reflect the different providers, something like /api/{provider}/frontends|backends instead of /api/frontends.  /api/frontends could then be an informational API that would return the merged config.

Btw, should I add a new commit, or ammend this one?